src/main/java/org/zalando/nakadi/service/EventPublisher.java

```diff
@@ -82,6 +95,17 @@ public EventPublishResult publish(final String events, final String eventTypeNam
         } catch (final EventPublishingException e) {
             LOG.error("error publishing event", e);
             return failed(batch);
+        } catch (InterruptedException | TimeoutException e) {
```

Comment:
I would better change timelineSync.workWithEventType() to throw some custom exception instead of InterruptedException and TimeoutException.